php/class-sync.php

```diff
@@ -198,18 +199,21 @@ public function is_synced( $post_id, $re_check = false ) {
 	 * @param mixed  $result        The result.
 	 */
 	public function log_sync_result( $attachment_id, $type, $result ) {
-		$log  = $this->managers['media']->get_post_meta( $attachment_id, self::META_KEYS['process_log'], true );
+		$log  = $this->managers['media']->get_process_logs( $attachment_id );
 		$keys = array_keys( $this->sync_base_struct );
 		if ( empty( $log ) || count( $log ) !== count( $keys ) ) {
 			$log = array_fill_keys( $keys, array() );
 		}
 		if ( isset( $log[ $type ] ) ) {
+			if ( is_wp_error( $result ) ) {
+				$result = $result->get_error_message();
+			}
```

Comment:
@pereirinha - To make detecting the result as an error, maybe we make `$result` an array like:
 ```php
 if ( is_wp_error( $result ) ) {
		$result = array(
				'error'   => $result->get_error_code(),
				'message' => $result->get_error_message(),
		);
}
```